pkg/health/checker/checker.go

```diff
@@ -213,28 +290,31 @@ func (c consulHealthChecker) WatchHealth(
 	}
 }
 
-func (c consulHealthChecker) WatchService(
+func (h healthChecker) WatchService(
 	serviceID string,
 	resultCh chan<- map[types.NodeName]health.Result,
 	errCh chan<- error,
 	quitCh <-chan struct{},
 	watchDelay time.Duration,
+	useHealthService bool,
+	status manifest.StatusStanza,
 ) {
 	defer close(resultCh)
-	var curIndex uint64 = 0
-
 	if watchDelay < time.Second {
 		watchDelay = time.Second
 	}
 
 	timer := time.NewTimer(0)
+	respChan := make(chan *hclient.HealthResponse, 1)
```

Comment:
this channel is never closed, which means we're going to leak resources every time `WatchService` is called. We need to make sure that it's closed once we expect no values to be sent on it anymore.

Long winded explanation warning: In my experience (and I think I read in a blog post somewhere), the best way to get channel closing right is to only have the sender on the channel do the closing. In this case, the sender is the `h.healthClient.HealthMonitor()` function, and the receiver is this function (`WatchService()`).

What this means is that I think that instead of declaring `respChan` here in this function, you should have`h.healthClient.HealthMonitor()` create it and return it. Then, `HealthMonitor()` should wait for the passed context to be canceled, and when it is, close the channel and exit the goroutine. This guarantees that no values are sent after the channel is closed (which causes a panic)

Reply:
What if there's a 
```golang
defer close(resultCh)
```
in HealthMonitor?